In [0]:
import numpy as np 
import pandas as pd 

In [0]:
dbutils.fs.unmount("/mnt/tokyoolympic1")


/mnt/tokyoolympic1 has been unmounted.


True

In [0]:
configs = {"fs.azure.account.auth.type": "OAuth",
"fs.azure.account.oauth.provider.type": "org.apache.hadoop.fs.azurebfs.oauth2.ClientCredsTokenProvider",
"fs.azure.account.oauth2.client.id": "ENTER CLIENT ID",
"fs.azure.account.oauth2.client.secret": 'ENTER SECRET KEY',
"fs.azure.account.oauth2.client.endpoint": "https://login.microsoftonline.com/ENTER END POINT /oauth2/token"}


dbutils.fs.mount(
source = "abfss://tokyo-olympic-data@projectdataolympics.dfs.core.windows.net", 
mount_point = "/mnt/tokyoolympic1",
extra_configs = configs
)
  

True

In [0]:
%fs
ls "/mnt/tokyoolympic1"

path name size modificationTime dbfs:/mnt/tokyoolympic1/raw-data/ raw-data/ 0 1737257693000 dbfs:/mnt/tokyoolympic1/transformed-data/ transformed-data/ 0 1737257712000

In [0]:
Athletes = pd.read_csv("/dbfs/mnt/tokyoolympic1/raw-data/athletes.csv", encoding='ISO-8859-1')
Coaches = pd.read_csv("/dbfs/mnt/tokyoolympic1/raw-data/coaches.csv", encoding='ISO-8859-1')
Gender = pd.read_csv("/dbfs/mnt/tokyoolympic1/raw-data/entries_gender.csv", encoding='ISO-8859-1')
Medals = pd.read_csv("/dbfs/mnt/tokyoolympic1/raw-data/medals.csv", encoding='ISO-8859-1')
Teams = pd.read_csv("/dbfs/mnt/tokyoolympic1/raw-data/teams.csv", encoding='ISO-8859-1')

In [0]:
print(Teams.head())
print("Number of rows and columns are :", Teams.shape)
print('/n')
print( "Info about Teams :")
print(Teams.info())
print("Number of unique Team :")
print(Teams.nunique())
print("number of duplicate rows :" ,Teams.duplicated().sum())
print()
print("number of missing values : " )
print(Teams.isnull().sum())

  TeamName      Discipline                     Country  Event
0  Belgium  3x3 Basketball                     Belgium    Men
1    China  3x3 Basketball  People's Republic of China    Men
2    China  3x3 Basketball  People's Republic of China  Women
3   France  3x3 Basketball                      France  Women
4    Italy  3x3 Basketball                       Italy  Women
Number of rows and columns are : (743, 4)
/n
Info about Teams :
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 743 entries, 0 to 742
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   TeamName    743 non-null    object
 1   Discipline  743 non-null    object
 2   Country     743 non-null    object
 3   Event       743 non-null    object
dtypes: object(4)
memory usage: 23.3+ KB
None
Number of unique Team :
TeamName      146
Discipline     20
Country        84
Event          36
dtype: int64
number of duplicate rows : 0

number of missing values : 
TeamNam

We observe that this dataset is composed of 743 rows and 4 columns. All the variables are of object type. The number of duplicate rows is zero, and there are also no missing values

In [0]:
print(Coaches.head())
print("Number of rows and columns are :",Coaches.shape)
print()
print( "Info about Coaches :")
print(Coaches.info())
print("Number of unique Coaches :")
print(Coaches.nunique())
print("number of duplicate rows :" ,Coaches.duplicated().sum())
print()
print("number of missing values : " )
print(Coaches.isnull().sum())

              Name        Country  Discipline Event
0  ABDELMAGID Wael          Egypt    Football   NaN
1        ABE Junya          Japan  Volleyball   NaN
2    ABE Katsuhiko          Japan  Basketball   NaN
3     ADAMA Cherif  Côte d'Ivoire    Football   NaN
4       AGEBA Yuya          Japan  Volleyball   NaN
Number of rows and columns are : (394, 4)

Info about Coaches :
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 394 entries, 0 to 393
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Name        394 non-null    object
 1   Country     394 non-null    object
 2   Discipline  394 non-null    object
 3   Event       249 non-null    object
dtypes: object(4)
memory usage: 12.4+ KB
None
Number of unique Coaches :
Name          381
Country        61
Discipline      9
Event           6
dtype: int64
number of duplicate rows : 1

number of missing values : 
Name            0
Country         0
Discipline      0
Event     

We observe that this dataset is composed of 394 rows and 4 columns. All the variables are of object type. We have one duplicate rows, and we have 145 missing values .

In [0]:
print(Athletes.head())
print("Number of rows and columns are :",Athletes.shape)
print()
print( "Info about Athletes :")
print(Athletes.info())
print("Number of unique Athletes :")
print(Athletes.nunique())
print("number of duplicate rows :" ,Athletes.duplicated().sum())
print()
print("number of missing values : " )
print(Athletes.isnull().sum())

          PersonName Country           Discipline
0    AALERUD Katrine  Norway         Cycling Road
1        ABAD Nestor   Spain  Artistic Gymnastics
2  ABAGNALE Giovanni   Italy               Rowing
3     ABALDE Alberto   Spain           Basketball
4      ABALDE Tamara   Spain           Basketball
Number of rows and columns are : (11085, 3)

Info about Athletes :
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11085 entries, 0 to 11084
Data columns (total 3 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   PersonName  11085 non-null  object
 1   Country     11085 non-null  object
 2   Discipline  11085 non-null  object
dtypes: object(3)
memory usage: 259.9+ KB
None
Number of unique Athletes :
PersonName    11062
Country         206
Discipline       46
dtype: int64
number of duplicate rows : 1

number of missing values : 
PersonName    0
Country       0
Discipline    0
dtype: int64


We observe that this dataset is composed of 11085 rows and 3 columns. All the variables are of object type. The number of duplicate rows is one, and there are also no missing values

In [0]:
print(Gender.head())
print("Number of rows and columns are :",Gender.shape)
print()
print( "Info about EntriesGender :")
print(Gender.info())
print("Number of unique EntriesGender:")
print(Gender.nunique())
print("number of duplicate rows :" ,Gender.duplicated().sum())
print()
print("number of missing values : " )
print(Gender.isnull().sum())

            Discipline  Female  Male  Total
0       3x3 Basketball      32    32     64
1              Archery      64    64    128
2  Artistic Gymnastics      98    98    196
3    Artistic Swimming     105     0    105
4            Athletics     969  1072   2041
Number of rows and columns are : (46, 4)

Info about EntriesGender :
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 46 entries, 0 to 45
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Discipline  46 non-null     object
 1   Female      46 non-null     int64 
 2   Male        46 non-null     int64 
 3   Total       46 non-null     int64 
dtypes: int64(3), object(1)
memory usage: 1.6+ KB
None
Number of unique EntriesGender:
Discipline    46
Female        38
Male          41
Total         41
dtype: int64
number of duplicate rows : 0

number of missing values : 
Discipline    0
Female        0
Male          0
Total         0
dtype: int64


We observe that this dataset is composed of 46 rows and 4 columns.Three variables are of int type and one variable is of object type. The number of duplicate rows is zero, and there are also no missing values.

In [0]:
print(Medals.head())
print("Number of rows and columns are :",Medals.shape)
print()
print( "Info about Medals :")
print(Medals.info())
print("Number of unique Medals:")
print(Medals.nunique())
print("number of duplicate rows :" ,Medals.duplicated().sum())
print()
print("number of missing values : " )
print(Medals.isnull().sum())

   Rank                 TeamCountry  Gold  Silver  Bronze  Total  Rank by Total
0     1    United States of America    39      41      33    113              1
1     2  People's Republic of China    38      32      18     88              2
2     3                       Japan    27      14      17     58              5
3     4               Great Britain    22      21      22     65              4
4     5                         ROC    20      28      23     71              3
Number of rows and columns are : (93, 7)

Info about Medals :
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 93 entries, 0 to 92
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Rank           93 non-null     int64 
 1   TeamCountry    93 non-null     object
 2   Gold           93 non-null     int64 
 3   Silver         93 non-null     int64 
 4   Bronze         93 non-null     int64 
 5   Total          93 non-null     int64 
 6   Rank by

We observe that this dataset is composed of 93 rows and 7 columns. Six variables are of int type and one variable is of object type. The number of duplicate rows is zero, and there are also no missing values.

In [0]:
import matplotlib.pyplot as plt
import plotly.express as px
data = Athletes.Discipline.value_counts()
num_disc = data.values
num_disc_index = data.index
plt.figure(figsize = (12,7))
fig= px.bar(data , x=num_disc_index, y=num_disc ,title="The most Discipline in the olympiade ",labels={"x":"Athletes","y":"Number"})
fig.update_xaxes(title_text="Discipline")
fig.show()

<Figure size 1200x700 with 0 Axes>

In [0]:
px.bar(Medals, x="TeamCountry", y="Total", color="Gold", title="Top Countries in terms of number of medals : ")

In [0]:
px.bar(Medals, x="TeamCountry", y="Gold",title="Gold Medals")

In [0]:
px.bar(Medals, x="TeamCountry", y="Silver" ,title="Silver Medals")

In [0]:
px.bar(Medals, x="TeamCountry", y="Bronze" ,title="Bronze Medals")

In [0]:
px.bar(Gender, x="Discipline", y="Total", color="Discipline", title="Total participants in each Discipline : ")

In [0]:
 px.bar(Gender, x="Discipline", y=["Female", "Male"], barmode="group"
             )

Report


The majority of athletes in the 2021 (2020) Tokyo Olympics hailed from the USA, Japan, and Australia. This dominance can be linked to their strong sporting cultures, substantial investment in sports infrastructure, and comprehensive talent development programs.

Athletics emerged as the most popular discipline in the Olympics, likely due to its universal appeal as a foundational sport that resonates with the core spirit of the Games. Its accessibility and wide reach attract participants and spectators from diverse backgrounds.

Japan leads in producing the most coaches, followed by the USA. This trend is reflective of the deeply ingrained sports culture in these nations. Athletes often begin their training at a young age, gaining valuable experience that positions them to transition into coaching roles later in their careers.

The USA achieved the highest number of Gold, Silver, and Bronze medals. This success stems from the country's significant investment in sports infrastructure, well-developed training programs, and extensive support systems for its athletes.







